src/locale.rs

```diff
@@ -61,28 +63,29 @@ unsafe fn lconv_to_locale(lconv: &libc::lconv) -> Locale {
 /// Read the numeric locale, or None on any failure.
 #[cfg(localeconv_l)]
 unsafe fn read_locale() -> Option<Locale> {
-    extern "C" {
-        fn localeconv_l(loc: libc::locale_t) -> *const libc::lconv;
+    unsafe extern "C" {
+        unsafe fn localeconv_l(loc: libc::locale_t) -> *const libc::lconv;
     }
 
     const empty: [libc::c_char; 1] = [0];
 
     // We create a new locale (pass 0 locale_t base)
     // and pass no "locale", so everything else is taken from the environment.
     // This is fine because we're only using this for numbers.
-    let loc = libc::newlocale(libc::LC_NUMERIC_MASK, empty.as_ptr(), 0 as libc::locale_t);
+    let loc =
+        unsafe { libc::newlocale(libc::LC_NUMERIC_MASK, empty.as_ptr(), 0 as libc::locale_t) };
     if loc.is_null() {
         return None;
     }
 
-    let lconv = localeconv_l(loc);
+    let lconv = unsafe { localeconv_l(loc) };
     let result = if lconv.is_null() {
         None
     } else {
-        Some(lconv_to_locale(&*lconv))
+        Some(unsafe { lconv_to_locale(&*lconv) })
```

Comment:
all our commits should try to pass all lints, to remove forward all lint-related references -- this allows readers to prune the search space for answers to the "what triggered this" question.